ml/src/main/scala/ai/h2o/sparkling/ml/params/HasIgnoredCols.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package ai.h2o.sparkling.ml.params
+
+import ai.h2o.sparkling.H2OFrame
+import ai.h2o.sparkling.ml.algos.H2OAlgoCommonUtils
+
+trait HasIgnoredCols extends H2OAlgoParamsBase with H2OAlgoCommonUtils {
+  private val ignoredCols =
+    new NullableStringArrayParam(this, "ignoredCols", "Names of columns to ignore for training.")
+
+  setDefault(ignoredCols -> null)
+
+  def getIgnoredCols(): Array[String] = $(ignoredCols)
+
+  def setIgnoredCols(value: Array[String]): this.type = set(ignoredCols, value)
+
+  override private[sparkling] def getH2OAlgorithmParams(trainingFrame: H2OFrame): Map[String, Any] = {
+    val newIgnoredCols = getIgnoredCols()
+    val existingMap = super.getH2OAlgorithmParams(trainingFrame)
+    val oldIgnoredCols = existingMap.getOrElse("ignored_columns", null).asInstanceOf[Array[String]]
+
+    if (oldIgnoredCols == null) {
+      existingMap + ("ignored_columns" -> newIgnoredCols)
+    } else if (newIgnoredCols == null) {
```

Comment:
Just to understand this, this bit is here because we alter ignored columns as part of HasRandomCols, is that correct?

Reply:
Yes, `randomCols` must be part of `ignoredCols`, otherwise `fit` method will throws a strange exception about a missing values in the weight column which is not set. H2O Python client does the same thing (`randomCols` -> `ignoredCols`)

Reply:
Thanks for the explanation!